samples/cpp/intelligent_scissors.cpp

```diff
@@ -0,0 +1,210 @@
+#include <iostream>
+#include <cmath>
+#include <string>
+#include <vector>
+#include <queue>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+using namespace cv;
+
+Mat img, img_pre_render, img_render;
+Point end;
+std::vector<std::vector<Point> > contours(1);
+std::vector<Point> tmp_contour;
+Mat zero_crossing, gradient_magnitude, Ix, Iy, hit_map_x, hit_map_y;
+
+struct Pix
+{
+    Point next_point;
+    double cost;
+
+    bool operator > (const Pix &b) const
+    {
+        return cost > b.cost;
+    }
+};
+
+static float local_cost(const Point& p, const Point& q)
+{
+    float fG = gradient_magnitude.at<float>(q.y, q.x);
+    float dp;
+    float dq;
+    bool isDiag = (p.x != q.x) && (p.y != q.y);
+
+    if ((Iy.at<float>(p) * (q.x - p.x) - Ix.at<float>(p) * (q.y - p.y)) >= 0)
+    {
+        dp = Iy.at<float>(p) * (q.x - p.x) - Ix.at<float>(p) * (q.y - p.y);
+        dq = Iy.at<float>(q) * (q.x - p.x) - Ix.at<float>(q) * (q.y - p.y);
+    }
+    else
+    {
+        dp = Iy.at<float>(p) * (p.x - q.x) + (-Ix.at<float>(p)) * (p.y - q.y);
+        dq = Iy.at<float>(q) * (p.x - q.x) + (-Ix.at<float>(q)) * (p.y - q.y);
+    }
+    if (isDiag)
+    {
+        dp /= sqrtf(2);
+        dq /= sqrtf(2);
+    }
+    else
+    {
+        fG /= sqrtf(2);
+    }
+    return  0.43 * zero_crossing.at<uchar>(q) + 0.43 * (acos(dp)
+            + acos(dq)) / acos(-1) + 0.14 * fG;
```

Comment:
```
C:\build\precommit_windows64\3.4\opencv\samples\cpp\intelligent_scissors.cpp(56): warning C4244: 'return': conversion from 'double' to 'float', possible loss of data [C:\build\precommit_windows64\build\samples\cpp\example_cpp_intelligent_scissors.vcxproj]
```
https://pullrequest.opencv.org/buildbot/builders/precommit_windows64/builds/23912/steps/compile%20release/logs/warnings%20%281%29

Reply:
I guess it should be something like
```cpp
return  0.43f * zero_crossing.at<uchar>(q) +
        0.43f * (acosf(dp) + acosf(dq)) / static_cast<float>(CV_PI) +
        0.14f * fG;
```

Is that correct that two coefficients are `0.43`?